trie/hexary.py

```diff
@@ -391,3 +422,16 @@ def __delitem__(self, key):
 
     def __contains__(self, key):
         return self.exists(key)
+
+    @contextlib.contextmanager
+    def squash_changes(self):
+        scratch_db = ScratchDB(self.db)
+        try:
+            with scratch_db.batch_commit(do_deletes=self.is_pruning):
+                memory_trie = type(self)(scratch_db, self.root_hash, prune=True)
+                yield memory_trie
+        except Exception as exc:
```

Comment:
I think that this whole `try/except/else` block can just be removed.  If an exception is raised, the statement which updates the `root_node` just won't be executed.